src/components/TimePicker.js

```diff
@@ -0,0 +1,482 @@
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {View} from 'react-native';
+import _ from 'underscore';
+import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
+import useLocalize from '@hooks/useLocalize';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import DateUtils from '@libs/DateUtils';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import styles from '@styles/styles';
+import * as StyleUtils from '@styles/StyleUtils';
+import CONST from '@src/CONST';
+import AmountTextInput from './AmountTextInput';
+import BigNumberPad from './BigNumberPad';
+import Button from './Button';
+import FormHelpMessage from './FormHelpMessage';
+import refPropTypes from './refPropTypes';
+import Text from './Text';
+
+const propTypes = {
+    /** Refs forwarded to the TextInputWithCurrencySymbol */
+    forwardedRef: refPropTypes,
+
+    /** Default value for the inputs */
+    defaultValue: PropTypes.string,
+
+    /** Callback to call when the Save button is pressed  */
+    onSubmit: PropTypes.func.isRequired,
+
+    /** Callback to call when the input changes */
+    onInputChange: PropTypes.func,
+};
+
+const defaultProps = {
+    forwardedRef: null,
+    onInputChange: () => {},
+    defaultValue: '',
+};
+
+const AMOUNT_VIEW_ID = 'amountView';
+const NUM_PAD_CONTAINER_VIEW_ID = 'numPadContainerView';
+const NUM_PAD_VIEW_ID = 'numPadView';
+
+function insertAtPosition(originalString, newSubstring, selectionPositionFrom, selectionPositionTo, fn) {
+    // Check for invalid positions
+    if (selectionPositionFrom < 0 || selectionPositionTo < 0 || selectionPositionFrom > originalString.length || selectionPositionTo > originalString.length) {
+        return;
+    }
+
+    // If the positions are the same, it means we're inserting at a point
+    if (selectionPositionFrom === selectionPositionTo) {
+        if (selectionPositionFrom === originalString.length) {
+            if (typeof fn === 'function') {
+                fn();
+            }
+            return originalString; // If the insertion point is at the end, simply return the original string
+        }
+        return originalString.slice(0, selectionPositionFrom) + newSubstring + originalString.slice(selectionPositionFrom);
+    }
+
+    // Replace the selected range
+    return originalString.slice(0, selectionPositionFrom) + newSubstring + originalString.slice(selectionPositionTo);
+}
+
+function decreaseBothSelectionByOne({start, end}) {
```

Comment:
Can we add docs to this file? Methods like this one are not very self explanatory IMO

Reply:
👍🏻